File: app/src/main/java/piuk/blockchain/android/ui/account/AccountView.java
package piuk.blockchain.android.ui.account;

import android.content.Intent;
import android.support.annotation.StringRes;

import info.blockchain.wallet.payload.data.LegacyAddress;

import piuk.blockchain.android.ui.base.View;
import piuk.blockchain.android.ui.customviews.ToastCustom;

interface AccountView extends View {

    void onShowTransferableLegacyFundsWarning(boolean isAutoPopup);

    void onSetTransferLegacyFundsMenuItemVisible(boolean visible);

    void showProgressDialog(@StringRes int message);

    void dismissProgressDialog();

    void onUpdateAccountsList();

    void showToast(@StringRes int message, @ToastCustom.ToastType String toastType);

    void broadcastIntent(Intent intent);

    void showWatchOnlyWarningDialog(String address);

    void showRenameImportedAddressDialog(LegacyAddress address);

    void startScanForResult();

    void showBip38PasswordDialog(String data);
}

File: app/src/main/java/piuk/blockchain/android/ui/account/AccountView.kt
package piuk.blockchain.android.ui.account

import android.content.Intent
import android.support.annotation.StringRes

import info.blockchain.wallet.payload.data.LegacyAddress

import piuk.blockchain.android.ui.base.View
import piuk.blockchain.android.ui.customviews.ToastCustom
import java.util.*

interface AccountView : View {

    val locale: Locale

    fun onShowTransferableLegacyFundsWarning(isAutoPopup: Boolean)

    fun onSetTransferLegacyFundsMenuItemVisible(visible: Boolean)

    fun showProgressDialog(@StringRes message: Int)

    fun dismissProgressDialog()

    fun showToast(@StringRes message: Int, @ToastCustom.ToastType toastType: String)

    fun broadcastIntent(intent: Intent)

    fun showWatchOnlyWarningDialog(address: String)

    fun showRenameImportedAddressDialog(address: LegacyAddress)

    fun startScanForResult()

    fun showBip38PasswordDialog(data: String)

    fun updateAccountList(displayAccounts: List<AccountItem>)

}
